src/Features/Core/Portable/Shared/TestHooks/Waiters/EditorAdornmentWaiter.cs

```diff
@@ -3,14 +3,14 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Windows.Threading;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Roslyn.Utilities;
 
 namespace Roslyn.Hosting.Diagnostics.Waiters
 {
-    internal abstract class EditorAdornmentWaiter : AsynchronousOperationListener
+    internal abstract class EditorAdornmentWaiter
```

Comment:
left these files since I wasn't sure what this is trying to accomplish. since it continues after task being completed. so the task itself shouldn't rely on UI thread being ran. so I wasn't sure what this Dispatcher is trying to do. if there is some work scheduled to UI thread, the original task should wait for it so, the chained dispatcher doesn't have any impact?